openedx/core/djangoapps/discussions/handlers.py

```diff
@@ -55,36 +55,44 @@ def update_course_discussion_config(configuration: CourseDiscussionConfiguration
     with transaction.atomic():
         log.info(f"Updating existing discussion topic links for {course_key}")
         for topic_link in DiscussionTopicLink.objects.filter(
-            context_key=course_key, provider_id=provider_id,
+            context_key=course_key,
+            provider_id=provider_id,
         ):
             lookup_key = topic_link.usage_key or topic_link.external_id
             topic_context = new_topic_map.pop(lookup_key, None)
-            # TODO: handle deleting topics that are no longer in use
-            # currently this will simply not work for course-wide topics since deleting the link will
-            # remove access to all posts in the topic.
```

Comment:
We are simply hiding such topics and not performing any deletion to avoid data loss.